lib/util.js

```diff
@@ -592,22 +593,21 @@ function formatValue(ctx, value, recurseTimes, ln) {
       } else if (keyLength === 0) {
         if (isExternal(value))
           return ctx.stylize('[External]', 'special');
-        return `${ctorName}{}`;
+        return `${tag}{}`;
       } else {
-        braces[0] = `${ctorName}{`;
+        braces[0] = `${tag}{`;
       }
     }
   }
 
   // Using an array here is actually better for the average case than using
-  // a Set. `seen` will only check for the depth and will never grow to large.
+  // a Set. `seen` will only check for the depth and will never grow too large.
   if (ctx.seen.indexOf(value) !== -1)
     return ctx.stylize('[Circular]', 'special');
 
   if (recurseTimes != null) {
     if (recurseTimes < 0)
-      return ctx.stylize(`[${constructor ? constructor.name : 'Object'}]`,
-                         'special');
+      return ctx.stylize(`[${tag.trim() || 'Object'}]`, 'special');
```

Comment:
We could avoid `trim()` here if we saved the original `tag`. Maybe change the current `tag` variable to `prefix` or something to differentiate between the two?